web/src/chartOptions/segmented/pcr.ts

```diff
@@ -1,52 +1,54 @@
 import {graphic} from "echarts/core";
-import {isEmpty} from "lodash";
+import {isEmpty, has, isNil} from "lodash";
+import {WgsCovPlotDB} from "../../db";
 
-export function getFluPrimerData(
-    {
-        sample,
-        db: {
-            segments,
-            primerMatches,
-            segCoords,
-        },
-    }) {
-    let primerFeature = [];
+export function getSegmentPrimerData(sample: string, db: WgsCovPlotDB) {
+    let primerFeatures = [];
+    let segments = Object.keys(db.segCoords); // trigger only segCoords is updated from get Dataset
     for (let segment of segments) {
-        let matchesToSampleSegment = primerMatches[sample][segment];
-        let coords = segCoords[segment];
-        for (let match of matchesToSampleSegment) {
-            let {
-                start,
-                end,
-                name,
-                query_aligned,
-                target_aligned,
-                matched_aligned,
-                cigar,
-                edit_distance,
-            } = match;
-            // TODO: push object instead of array? does echarts need value to be an array instead of object?
-            primerFeature.push({
-                value: [
-                    start + coords.start,
-                    end + coords.start,
-                    10, // height for primer annotation in log scale
+        let matchesData;
+        if (has(db.primer_matches, [sample, segment])) {
+            matchesData = db.primer_matches[sample][segment];
+        }
+        let coords = db.segCoords[segment]
+        if (!isNil(matchesData)) {
+            for (let match of matchesData) {
+                let {
+                    start,
+                    end,
                     name,
                     query_aligned,
                     target_aligned,
                     matched_aligned,
                     cigar,
                     edit_distance,
-                ],
-                itemStyle: {color: "#b71ae3"}
-            });
+                    other_locations,
+                } = match;
+                // TODO: push object instead of array? does echarts need value to be an array instead of object?
```

Comment:
@nhhaidee There are a lot of values being packed into the array for primer features. Does that information need to be an array? Can it be an object/dict with informative key names instead? Also can the necessary data structure be created on the Python side instead of having to recompute things in JS?